hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieHFileDataBlock.java

```diff
@@ -175,7 +175,26 @@ protected <T> ClosableIterator<HoodieRecord<T>> deserializeRecords(byte[] conten
     FileSystem fs = FSUtils.getFs(pathForReader.toString(), FSUtils.buildInlineConf(getBlockContentLocation().get().getHadoopConf()));
     // Read the content
     HoodieAvroHFileReader reader = new HoodieAvroHFileReader(fs, pathForReader, content, Option.of(getSchemaFromHeader()));
-    return unsafeCast(reader.getRecordIterator(readerSchema));
+
+    ClosableIterator<HoodieRecord<IndexedRecord>> recordIterator = reader.getRecordIterator(readerSchema);
+    ClosableIterator<HoodieRecord<IndexedRecord>> iterator = new ClosableIterator<HoodieRecord<IndexedRecord>>() {
+      @Override
+      public void close() {
+        recordIterator.close();
+        reader.close();
```

Comment:
Isn't the `recordIterator.close()` just closing the reader? What nested in another iterator?